sdk/ml/azure-ai-ml/azure/ai/ml/entities/_resource.py

```diff
@@ -76,7 +76,7 @@ def __init__(
         # Hide read only properties in kwargs
         self._id = kwargs.pop("id", None)
         self.__source_path: Optional[str] = kwargs.pop("source_path", None)
-        self._base_path = kwargs.pop("base_path", os.getcwd())
+        self._base_path = kwargs.pop("base_path", None) or os.getcwd()  # base path should never be None
```

Comment:
seems two form but same logic to me?

why change?

Reply:
if `kwargs["base_path"]` is `None`, it will be `None` previously, but will be `os.getcwd()` after this change. This is the root cause of our ASK MODE bug

Reply:
got it